apps/astarte_appengine_api/lib/astarte_appengine_api_web/metrics/appengine_instrumenter.ex

```diff
@@ -0,0 +1,103 @@
+#
+# This file is part of Astarte.
+#
+# Copyright 2020 Ispirata Srl
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+defmodule Astarte.AppEngine.APIWeb.Metrics.AppEngineInstrumenter do
+  use Prometheus.Metric
+
+  @sent_messages_specs [
+    name: :sent_messages_total,
+    help: "AppEngine sent messages count",
+    labels: [:realm]
+  ]
+
+  @exchanged_bytes_specs [
+    name: :exchanged_bytes_total,
+    help: "AppEngine exchanged bytes count",
+    labels: [:realm]
+  ]
+
+  @opened_rooms_specs [
+    name: :opened_rooms,
+    help: "AppEngine opened rooms count",
+    labels: [:realm]
+  ]
+
+  @sent_events_specs [
+    name: :sent_events_total,
+    help: "AppEngine sent events count",
+    labels: [:realm]
+  ]
+
+  def setup do
+    Counter.declare(@sent_messages_specs)
+    Counter.declare(@exchanged_bytes_specs)
+    Gauge.declare(@opened_rooms_specs)
+    Counter.declare(@sent_events_specs)
+
+    events = [
+      [:astarte, :appengine, :device, :message_sent],
+      [:astarte, :appengine, :channels, :room_opened],
+      [:astarte, :appengine, :channels, :room_closed],
+      [:astarte, :appengine, :channels, :event_sent]
```

Comment:
we sill miss a consumed message event, when a message is consumed from the rabbitmq queue and maybe a further event when a message is discarded because the room is not open anymore.